cardano-api/golden/Test/Golden/ErrorsSpec.hs

```diff
@@ -295,3 +302,15 @@ spec = describe "Test.Golden.Errors" $ do
   it "TxMetadataRangeError" $ H.requireTest $ do
     flip H.diffVsGoldenFile "golden/files/errors/TxMetadataRangeError/TxMetadataJsonToplevelNotMap.txt"
       $ displayError $ TxMetadataBytesTooLong 0
+
+testErrorMessage :: (HasCallStack, MonadIO m, MonadTest m, Data p , Error p) => p -> m ()
+testErrorMessage err = withFrozenCallStack $ do
+  let typeName = show $ typeOf err
+      constructorName = show $ toConstr err
+  H.note_ "Incorrect error message in golden file"
+  displayError err `H.diffVsGoldenFile` ("golden/files/errors/" <> typeName <> "/" <> constructorName <> ".txt")
+
+deriving instance Typeable ProtocolParametersConversionError
+deriving instance Data ProtocolParametersConversionError
```

Comment:
I prefer if the above two instances lived with the type.  Is there any reason it shouldn't?